# Draft analysis 

---

Group name: Group C, Burcin Aksel & Nico Henzel

---


## Introduction

### Inhalte aus Project Proposal übernehmen und konsolidieren
### Burcin

*This section includes an introduction to the project motivation, data, and research question. Include a data dictionary* 

## Setup

### Anpassen, sobald Import benötigt wird
### Nico

In [1]:
import pandas as pd
import numpy as np
import altair as alt
import matplotlib as mlp
import xlrd # needed to read excel files. install xlrd module if import does not work!
import shutil # needed to copy files


## Data

## Nico

### Inhalte aus Data Correction und Project Proposal übernehmen

## Import data

In [2]:
# Declare variables
external_data = '..\\data\\external\\'
raw_data = '..\\data\\raw\\'

# File names
orig_income_file = 'Median_Household_Income_By_State_1990-2017.xls'
target_income_file = 'Median_Household_Income_By_State_1990-2017.csv'
orig_death_file = 'NCHS_-_Leading_Causes_of_Death__United_States.csv'


In [3]:
# Save external median income data as csv in folder 'raw'
# Read file
xls_household_file = pd.read_excel(external_data+orig_income_file)
# Save file
xls_household_file.to_csv(raw_data+target_income_file,index = None, header=True)
# Copy external leading cause of death data into folder 'raw'
shutil.copy(external_data+orig_death_file, raw_data+orig_death_file)


'..\\data\\raw\\NCHS_-_Leading_Causes_of_Death__United_States.csv'

In [17]:
# Declare both dataframes
df_income = pd.read_csv(raw_data+target_income_file)
df_death = pd.read_csv(raw_data+orig_death_file)

### Data structure

In [12]:
df_income.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 68 entries, 0 to 67
Data columns (total 32 columns):
 #   Column                                                                              Non-Null Count  Dtype  
---  ------                                                                              --------------  -----  
 0   Table 102.30. Median household income, by state: Selected years, 1990 through 2017  59 non-null     object 
 1   Unnamed: 1                                                                          54 non-null     object 
 2   Unnamed: 2                                                                          54 non-null     object 
 3   Unnamed: 3                                                                          54 non-null     float64
 4   Unnamed: 4                                                                          52 non-null     float64
 5   Unnamed: 5                                                                          54 non-null     f

In [13]:
df_income

Table 102.30. Median household income, by state: Selected years, 1990 through 2017  \
0   [In constant 2017 dollars. Standard errors app...                                   
1                                               State                                   
2                                                   1                                   
3                           United States ...........                                   
4                        Alabama ....................                                   
..                                                ...                                   
63                         Wyoming ..................                                   
64  \1\Based on 1989 incomes collected in the 1990...                                   
65  \2\Based on 1999 incomes collected in the 2000...                                   
66  NOTE: Constant dollars adjusted by the Consume...                                   
67  SOURCE: U.S. Department of Commerce, Census Bu...                                   

   Unnamed: 1 Unnamed: 2  Unnamed: 3  Unnamed: 4  Unnamed: 5  Unnamed: 6  \
0         NaN        NaN         NaN         NaN         NaN         NaN   
1     1990\1\    2000\2\      2005.0         NaN      2010.0         NaN   
2           2          3         4.0         NaN         5.0         NaN   
3       57500      62000     58200.0        80.0     56400.0        40.0   
4       45200      50400     46400.0       400.0     45600.0       320.0   
..        ...        ...         ...         ...         ...         ...   
63      51900      55900     58100.0      1160.0     60300.0      1300.0   
64        NaN        NaN         NaN         NaN         NaN         NaN   
65        NaN        NaN         NaN         NaN         NaN         NaN   
66        NaN        NaN         NaN         NaN         NaN         NaN   
67        NaN        NaN         NaN         NaN         NaN         NaN   

    Unnamed: 7  Unnamed: 8  Unnamed: 9  ...  Unnamed: 22  Unnamed: 23  \
0          NaN         NaN         NaN  ...          NaN          NaN   
1       2013.0         NaN      2014.0  ...          NaN          NaN   
2          6.0         NaN         7.0  ...          NaN          NaN   
3      55100.0        40.0     55600.0  ...          NaN          NaN   
4      45200.0       410.0     44400.0  ...          NaN          NaN   
..         ...         ...         ...  ...          ...          ...   
63     61900.0      1150.0     59100.0  ...          NaN          NaN   
64         NaN         NaN         NaN  ...          NaN          NaN   
65         NaN         NaN         NaN  ...          NaN          NaN   
66         NaN         NaN         NaN  ...          NaN          NaN   
67         NaN         NaN         NaN  ...          NaN          NaN   

    Unnamed: 24  Unnamed: 25  Unnamed: 26  Unnamed: 27  Unnamed: 28  \
0           NaN          NaN          NaN          NaN          NaN   
1           NaN          NaN          NaN          NaN          NaN   
2           NaN          NaN          NaN          NaN          NaN   
3           NaN          NaN          NaN          NaN          NaN   
4           NaN          NaN          NaN          NaN          NaN   
..          ...          ...          ...          ...          ...   
63          NaN          NaN          NaN          NaN          NaN   
64          NaN          NaN          NaN          NaN          NaN   
65          NaN          NaN          NaN          NaN          NaN   
66          NaN          NaN          NaN          NaN          NaN   
67          NaN          NaN          NaN          NaN          NaN   

    Unnamed: 29  Unnamed: 30  Unnamed: 31  
0           NaN          NaN          NaN  
1           NaN          NaN          NaN  
2           NaN          NaN               
3           NaN          NaN          NaN  
4           NaN          NaN          NaN  
..          ...          ..

In [14]:
df_death.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10868 entries, 0 to 10867
Data columns (total 6 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Year                     10868 non-null  int64  
 1   113 Cause Name           10868 non-null  object 
 2   Cause Name               10868 non-null  object 
 3   State                    10868 non-null  object 
 4   Deaths                   10868 non-null  int64  
 5   Age-adjusted Death Rate  10868 non-null  float64
dtypes: float64(1), int64(2), object(3)
memory usage: 509.6+ KB


In [15]:
df_death

Year                                     113 Cause Name  \
0      2017  Accidents (unintentional injuries) (V01-X59,Y8...   
1      2017  Accidents (unintentional injuries) (V01-X59,Y8...   
2      2017  Accidents (unintentional injuries) (V01-X59,Y8...   
3      2017  Accidents (unintentional injuries) (V01-X59,Y8...   
4      2017  Accidents (unintentional injuries) (V01-X59,Y8...   
...     ...                                                ...   
10863  1999  Nephritis, nephrotic syndrome and nephrosis (N...   
10864  1999  Nephritis, nephrotic syndrome and nephrosis (N...   
10865  1999  Nephritis, nephrotic syndrome and nephrosis (N...   
10866  1999  Nephritis, nephrotic syndrome and nephrosis (N...   
10867  1999  Nephritis, nephrotic syndrome and nephrosis (N...   

                   Cause Name          State  Deaths  Age-adjusted Death Rate  
0      Unintentional injuries  United States  169936                     49.4  
1      Unintentional injuries        Alabama    2703                     53.8  
2      Unintentional injuries         Alaska     436                     63.7  
3      Unintentional injuries        Arizona    4184                     56.2  
4      Unintentional injuries       Arkansas    1625                     51.8  
...                       ...            ...     ...                      ...  
10863          Kidney disease       Virginia    1035                     16.9  
10864          Kidney disease     Washington     278                      5.2  
10865          Kidney disease  West Virginia     345                     16.4  
10866          Kidney disease      Wisconsin     677                     11.9  
10867          Kidney disease        Wyoming      30                      6.8  

[10868 rows x 6 columns]

In the death dataset we have 10868 cases and 6 columns.

The columns are:
* years, from 1999 to 2017
* 113 cause name, the NDI ICD-10 113 categories for causes of death
* cause name, the generic name for the death cause defined in the 113 cause name column
* state, in which state the data was collected
* death, the count of the total deaths 
* age-adjusted death rate, the standardized death percentage for the specific state in the observed year.


The following steps are:

* The income dataset needs to be cleaned and corrected in order to be joined with the death dataset (see data-corrections)
* The columns for the income dataset need to be transformed in order to have the same format as the death dataset.
* The join can be done by state, the missing income values for the years not represented in the income dataset need to either be dropped or replaced


### Data corrections

#### Income Dataset

In [31]:
df_income.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 68 entries, 0 to 67
Data columns (total 32 columns):
 #   Column                                                                              Non-Null Count  Dtype  
---  ------                                                                              --------------  -----  
 0   Table 102.30. Median household income, by state: Selected years, 1990 through 2017  59 non-null     object 
 1   Unnamed: 1                                                                          54 non-null     object 
 2   Unnamed: 2                                                                          54 non-null     object 
 3   Unnamed: 3                                                                          54 non-null     float64
 4   Unnamed: 4                                                                          52 non-null     float64
 5   Unnamed: 5                                                                          54 non-null     f

We will perform different corrections for this dataset. The steps are explained as comments within the code

In [24]:
# We only need the first 16 columns 
# and we can also drop the columns 4,6,8,10,12 and 14 since they only show NaN values
column_lst = [0,1,2,3,5,7,9,11,13,15]
df_income_corrected = df_income[df_income.columns[column_lst]]
# We don't need row 0,2 and row 64-67
row_drop_lst = [0,2,64,65,66,67]
df_income_corrected = df_income_corrected.drop(row_drop_lst)
# There are a few rows with empty entries left, which we can get rid off 
# since those rows have no state assigned to it, they are only used as seperators
df_income_corrected.dropna(inplace=True)
# The column names are actually in the first row, additionally they need to be adjusted
column_names = ['State','1990','2000','2005','2010','2013','2014','2015','2016','2017']
df_income_corrected.columns = column_names
# Row number 1 can be dropped
df_income_corrected.drop(1,inplace=True)
# The dots in the state column can be removed
df_income_corrected = df_income_corrected.replace(r'\.','',regex=True)
# Lastly we reset the row index drop the old index 
df_income_corrected.reset_index(drop = True, inplace = True)


In [25]:
df_income_corrected

State   1990   2000     2005     2010     2013     2014  \
0          United States   57500  62000  58200.0  56400.0  55100.0  55600.0   
1                Alabama   45200  50400  46400.0  45600.0  45200.0  44400.0   
2                 Alaska   79300  76100  70800.0  72800.0  76100.0  74200.0   
3                Arizona   52700  59900  55700.0  52700.0  51100.0  51900.0   
4               Arkansas   40500  47500  44000.0  43200.0  42700.0  42800.0   
5             California   68500  70100  67500.0  65000.0  63400.0  64200.0   
6               Colorado   57700  69700  63700.0  60900.0  62000.0  63500.0   
7            Connecticut   79900  79600  76700.0  72100.0  70700.0  72600.0   
8               Delaware   66800  69900  66100.0  62900.0  61000.0  61900.0   
9   District of Columbia   58800  59200  59400.0  68600.0  71200.0  74300.0   
10               Florida   52600  57300  53400.0  50000.0  48500.0  49200.0   
11               Georgia   55500  62600  57400.0  52300.0  50400.0  51100.0   
12                Hawaii   74300  73500  73100.0  71000.0  71700.0  72100.0   
13                 Idaho   48300  55500  52100.0  49000.0  49300.0  49600.0   
14              Illinois   61700  68800  63200.0  59700.0  59200.0  59500.0   
15               Indiana   55100  61300  55400.0  50300.0  50100.0  51200.0   
16                  Iowa   50200  58300  54900.0  54000.0  55100.0  55700.0   
17                Kansas   52200  60000  54000.0  54400.0  53700.0  54400.0   
18              Kentucky   43100  49700  47000.0  45100.0  45700.0  44500.0   
19             Louisiana   42000  48100  46200.0  47900.0  46500.0  46200.0   
20                 Maine   53300  55000  53900.0  51600.0  49500.0  51300.0   
21              Maryland   75400  78000  77500.0  77600.0  76400.0  76700.0   
22         Massachusetts   70700  74500  72000.0  69900.0  70400.0  71700.0   
23              Michigan   59400  65900  57900.0  51200.0  50900.0  51700.0   
24             Minnesota   59200  69500  65500.0  62500.0  64000.0  63700.0   
25           Mississippi   38500  46200  41400.0  41500.0  40000.0  41100.0   
26              Missouri   50500  56000  52800.0  49900.0  49500.0  50100.0   
27               Montana   44000  48700  49500.0  48100.0  49500.0  48000.0   
28              Nebraska   49800  57900  55200.0  54500.0  54200.0  54600.0   
29                Nevada   59400  65800  61900.0  57500.0  54000.0  53300.0   
30         New Hampshire   69500  73000  71400.0  68800.0  67700.0  69000.0   
31            New Jersey   78300  81400  77600.0  76300.0  74000.0  74500.0   
32            New Mexico   46100  50400  47200.0  47400.0  46200.0  46400.0   
33              New York   63100  64000  62300.0  61000.0  60500.0  61000.0   
34        North Carolina   51000  57800  51200.0  48800.0  48400.0  48300.0   
35          North Dakota   44400  51100  51600.0  54800.0  58800.0  61200.0   
36                  Ohio   54900  60400  54700.0  50800.0  50700.0  51100.0   
37              Oklahoma   45100  49300  46600.0  47400.0  48200.0  49300.0   
38                Oregon   52200  60400  54000.0  52500.0  53000.0  52900.0   
39          Pennsylvania   55600  59200  56000.0  55500.0  54800.0  55200.0   
40          Rhode Island   61600  62100  64700.0  58900.0  58900.0  56900.0   
41        South Carolina   50300  54700  49500.0  47300.0  46500.0  46900.0   
42          South Dakota   43100  52100  50700.0  51700.0  51600.0  52800.0   
43             Tennessee   47500  53700  48900.0  46700.0  46700.0  46000.0   
44                 Texas   51700  58900  53000.0  54800.0  54500.0  55000.0   
45                  Utah   56400  67500  60300.0  61700.0  63000.0  63100.0   
46               Vermont   57000  60300  57500.0  55700.0  55400.0  56100.0   
47              Virginia   63800  68900  68200.0  68400.0  66100.0  67300.0   
48            Washington   59700  67600  62000.0  62700.0  61600.0  63600.0   
49         West Virginia   39800  43800  42100.0  43100.0  43500.0  42

We also need to make some data transformations in order for this dataset to be joined later on.

In [26]:
# We transform the table to show the median income for a state in a single year
# We use the melt() function for this
lst_years = ['1990','2000','2005','2010','2013','2014','2015','2016','2017']
df_income_corrected = df_income_corrected.melt(
    id_vars= ['State'], 
    value_vars= lst_years, 
    var_name= 'year', 
    value_name= 'median_household_income' 
    )

# The State column should be lowercase
df_income_corrected = df_income_corrected.rename(columns = {
    'State':'state'
})
# The types need to be declared, state holds categorial values, year has integers and income holds float numbers
df_income_corrected['state'] = df_income_corrected['state'].astype('category')
df_income_corrected['year'] = df_income_corrected['year'].astype('int')
df_income_corrected['median_household_income'] = df_income_corrected['median_household_income'].astype('float')

In [32]:
df_income_corrected.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 468 entries, 0 to 467
Data columns (total 3 columns):
 #   Column                   Non-Null Count  Dtype   
---  ------                   --------------  -----   
 0   state                    468 non-null    category
 1   year                     468 non-null    int32   
 2   median_household_income  468 non-null    float64 
dtypes: category(1), float64(1), int32(1)
memory usage: 8.5 KB


In [28]:
file = 'Corrected_Median_Household_Income_By_State_1990-2017.csv'
df_income_corrected.to_csv(raw_data+file,index = None, header=True)

We prepared the income data set in order to use it for our project. 
The original .xls data can be found in the data folder under external or the .csv under raw 

In the final death data set we have 468 cases and 3 columns.

The columns are:
* state, in which the data was collected
* years, 1990, 2000, 2005, 2010 and 2013-2017
* median_household_income

#### Death Dataset

In [29]:
df_death.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10868 entries, 0 to 10867
Data columns (total 6 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Year                     10868 non-null  int64  
 1   113 Cause Name           10868 non-null  object 
 2   Cause Name               10868 non-null  object 
 3   State                    10868 non-null  object 
 4   Deaths                   10868 non-null  int64  
 5   Age-adjusted Death Rate  10868 non-null  float64
dtypes: float64(1), int64(2), object(3)
memory usage: 509.6+ KB


* The Dtypes 113 Cause Name, Cause Name and State need to be changed to category.
* The year, deaths and death rate column already have the right typing.
* Also, there are no missing values present in the dataset.


In [16]:
# JOIN

After joining, the median_household_income will be an additional column for the death data set.

This will leave us with 6 columns in total, which we can use for our data analysis (we will disregard the 113 Cause name column since the same information is provided by the cause name column)

### Variable lists
## Burcin

In [ ]:
# auf Fertigem Datensatz Variablen auflisten

### Data splitting

## Burcin

## Analysis

## Nico

Analysis Approach umsetzen

### Descriptive statistics

### Exploratory data analysis

### Relationships

## Model

## Nico
### Kopieren aus Notebook 44 und Modeltypen passend zu proposal umsetzen

Abstimmung, welche Metriken und Erkenntnisse aus den Modellen für uns relevant sind

### Select model

### Training and validation

### Fit model

### Evaluation on test set

### Save model



Save your model in the folder `models/`. Use a meaningful name and a timestamp.

## Conclusions

Am Ende die weiteren Schritte und bisherigen Interprationen auflisten.